community/command-line/src/main/java/org/neo4j/commandline/admin/Usage.java

```diff
@@ -40,75 +39,53 @@ public Usage( String scriptName, CommandLocator commands )
+    {
+        final CommandUsage commandUsage = new CommandUsage( command, scriptName );
+        commandUsage.printDetailed( output );
+    }
+
     public void print( Consumer<String> output )
     {
         output.accept( format( "usage: %s <command>", scriptName ) );
         output.accept( "" );
         output.accept( "Manage your Neo4j instance." );
         output.accept( "" );
+
         output.accept( "environment variables:" );
-        output.accept( "    NEO4J_DEBUG   Set to anything to enable debug output." );
-        output.accept( "    NEO4J_HOME    Neo4j home directory." );
-        output.accept( "    NEO4J_CONF    Path to directory which contains neo4j.conf." );
-        output.accept( "" );
+        printEnvironmentVariables( output );
+
         output.accept( "available commands:" );
-        printCommandsUnderASegment( output );
+        printCommands( output );
+
         output.accept( "" );
         output.accept( format( "Use %s help <command> for more details.", scriptName ) );
     }
 
-    private void printCommandsUnderASegment( Consumer<String> output )
+    private void printEnvironmentVariables( Consumer<String> output )
     {
-        List<AdminCommand.Provider> providers = new ArrayList<>();
-        commands.getAllProviders().forEach( providers::add );
-        providers.sort( Comparator.comparing( AdminCommand.Provider::name ) );
-        providers.forEach( command ->
-        {
-            final CommandUsage commandUsage = new CommandUsage( command, scriptName );
-            commandUsage.printIndentedSummary( output );
-        } );
-    }
-
-    public void printUsageForCommand( AdminCommand.Provider command, Consumer<String> output )
-    {
-        final CommandUsage commandUsage = new CommandUsage( command, scriptName );
-        commandUsage.printDetailed( output );
+        output.accept( "    NEO4J_CONF    Path to directory which contains neo4j.conf." );
+        output.accept( "    NEO4J_DEBUG   Set to anything to enable debug output." );
+        output.accept( "    NEO4J_HOME    Neo4j home directory." );
+        output.accept( "" );
     }
 
-    public static class CommandUsage
+    private void printCommands( Consumer<String> output )
     {
-        private final AdminCommand.Provider command;
-        private final String scriptName;
-
-        public CommandUsage( AdminCommand.Provider command, String scriptName )
-        {
-            this.command = command;
-            this.scriptName = scriptName;
-        }
-
-        public void printSummary( Consumer<String> output )
-        {
-            output.accept( format( "%s", command.name() ) );
-            output.accept( "    " + command.summary() );
-        }
-
-        public void printIndentedSummary( Consumer<String> output )
-        {
-            printSummary( s -> output.accept( "    " + s ) );
-        }
+        Map<AdminCommandSection,List<AdminCommand.Provider>> groupedProviders = groupProvidersBySegment();
```

Comment:
There seem to be a mix of terminology. Could we keep it to `Section` everywhere?